plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -114,6 +114,8 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 			Pod:              pod,
 			FailedPredicates: failedPredicateMap,
 		}
+	} else if len(filteredNodes) == 1 {
+		return filteredNodes[0].Name, nil
```

Comment:
You idea make sense. But what I'm not sure is whether this will improve scheduling. Maybe could you please post a simple test?

Reply:
Do you means adding some testcase in generic_scheduler_test.go?  If there is only one node in k8s cluster, it will definatly improve sheduling :).

Reply:
I'm referring a simple benchmark, you can just post it in comment. This will help reviewers to know if this fix is worthy.

Reply:
Sorry, I really haven't got a clue about benchmark. Can you give me a very simple example here? Thanks.

Reply:
@malc0lm You should prove your change can improve the scheduling time, for example, benchmark the scheduling latency with/ & without your patch. (which I actually doubt will be much difference).

You can see @wojtek-t also show the same concern here: https://github.com/kubernetes/kubernetes/pull/51334#issuecomment-325347775